laura_exercises.ipynb

```diff
@@ -0,0 +1,206 @@
+    "    return titles\n",
+    "\n",
+    "def parse_bricks(url):\n",
+    "    \"\"\" Fetches Lego Bricks page and extracts titles \"\"\"\n",
+    "    # Lookup the documentation to the \"requests\" library\n",
+    "    #\n",
+    "    # Use requests to make a get request to the\n",
+    "    # url given in the argument \"url\" (which is a string)\n",
+    "    # and get the raw HTML body of the response\n",
+    "    #\n",
+    "    # Use \"BeautifulSoup\" to parse this HTML. \n",
+    "    #\n",
+    "    # Use the \"get_titles\" function to extract the\n",
+    "    # titles from the BeautifulSoup object.\n",
+    "    #\n",
+    "    # Return the titles\n",
+    "    request = requests.get(url)\n",
+    "    soup = BeautifulSoup(request.text, 'html.parser')\n",
+    "    titles = get_titles(soup)\n",
+    "    return titles\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "bricks = parse_bricks(START_URL)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "assert(bricks[0] == '10251:  Brick Bank')\n",
+    "assert(bricks[9] == '10722:  Snake Showdown')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Exercise #2\n",
+    "\n",
+    "# Now write code that gets you all the links from ALL the pages.\n",
+    "\n",
+    "# HINT: you will probably want to extract the URL in the \"next\" button on \n",
+    "# the bottom of the search pagination, which looks like \">\".\n",
+    "\n",
+    "# HINT HINT: Think of the previous exercise on API's and internet data.\n",
+    "# The Pokemon API returned JSON, that we converted to a dictionary, that\n",
+    "# had a nice structure. In particular, there were two top-level keys of interest, \n",
+    "# one had the \"results\" in a list, the other was the \"next\" url to call to get\n",
+    "# more items. If you can replicate this return structure, you will be able to \n",
+    "# almost reuse the while loop you had there!\n",
+    "\n",
+    "# HINT HINT HINT: There's no reason you shouldn't be able to reuse the previous \n",
+    "# functions (get_titles and parse_bricks)\n",
+    "\n",
+    "def next_page(soup):\n",
+    "    pages = soup.find('div', attrs = {'class':'pagination'})\n",
+    "    next_page = pages.find('li', attrs = {'class':'next'})\n",
+    "    next_page_link = next_page.find('a')\n",
```

Comment:
Fine, but again, why not use CSS notation that we went over? `'div.pagination li.next a'`